storage/factory/pruningStorerFactory.go

```diff
@@ -0,0 +1,564 @@
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+)
+
+var log = logger.GetOrCreate("storage/factory")
+
+// StorageServiceFactory handles the creation of storage services for both meta and shards
+type StorageServiceFactory struct {
+	generalConfig      *config.Config
+	shardCoordinator   sharding.Coordinator
+	uniqueID           string
+	epochStartNotifier storage.EpochStartNotifier
+}
+
+// NewStorageServiceFactory will return a new instance of StorageServiceFactory
+func NewStorageServiceFactory(
+	config *config.Config,
+	shardCoordinator sharding.Coordinator,
+	uniqueID string,
+	epochStartNotifier storage.EpochStartNotifier,
+) (*StorageServiceFactory, error) {
+	if config == nil {
+		return nil, storage.ErrNilConfig
+	}
+	if check.IfNil(shardCoordinator) {
+		return nil, storage.ErrNilShardCoordinator
+	}
+	if len(uniqueID) == 0 {
+		return nil, storage.ErrEmptyUniqueIdentifier
+	}
+	if check.IfNil(epochStartNotifier) {
+		return nil, storage.ErrNilEpochStartNotifier
+	}
+
+	return &StorageServiceFactory{
+		generalConfig:      config,
+		shardCoordinator:   shardCoordinator,
+		uniqueID:           uniqueID,
+		epochStartNotifier: epochStartNotifier,
+	}, nil
+}
+
+// CreateForShard will return the storage service which contains all storers needed for a shard
+func (psf *StorageServiceFactory) CreateForShard() (dataRetriever.StorageService, error) {
+	var headerUnit *pruning.PruningStorer
+	var peerBlockUnit *pruning.PruningStorer
+	var miniBlockUnit *pruning.PruningStorer
+	var txUnit *pruning.PruningStorer
+	var metachainHeaderUnit *pruning.PruningStorer
+	var unsignedTxUnit *pruning.PruningStorer
+	var rewardTxUnit *pruning.PruningStorer
+	var metaHdrHashNonceUnit *pruning.PruningStorer
+	var shardHdrHashNonceUnit *pruning.PruningStorer
+	var bootstrapUnit *pruning.PruningStorer
+	var err error
+
+	defer func() {
+		// cleanup
+		if err != nil {
+			if headerUnit != nil {
+				_ = headerUnit.DestroyUnit()
+			}
+			if peerBlockUnit != nil {
+				_ = peerBlockUnit.DestroyUnit()
+			}
+			if miniBlockUnit != nil {
+				_ = miniBlockUnit.DestroyUnit()
+			}
+			if txUnit != nil {
+				_ = txUnit.DestroyUnit()
+			}
+			if unsignedTxUnit != nil {
+				_ = unsignedTxUnit.DestroyUnit()
+			}
+			if rewardTxUnit != nil {
+				_ = rewardTxUnit.DestroyUnit()
+			}
+			if metachainHeaderUnit != nil {
+				_ = metachainHeaderUnit.DestroyUnit()
+			}
+			if metaHdrHashNonceUnit != nil {
+				_ = metaHdrHashNonceUnit.DestroyUnit()
+			}
+			if shardHdrHashNonceUnit != nil {
+				_ = shardHdrHashNonceUnit.DestroyUnit()
+			}
+			if bootstrapUnit != nil {
+				_ = bootstrapUnit.DestroyUnit()
+			}
+		}
+	}()
+
+	fullArchiveMode := psf.generalConfig.StoragePruning.FullArchive
+	numOfEpochsToKeep := uint32(psf.generalConfig.StoragePruning.NumEpochsToKeep)
+	numOfActivePersisters := uint32(psf.generalConfig.StoragePruning.NumActivePersisters)
+
+	txUnitStorerArgs := &pruning.PruningStorerArgs{
+		Identifier:            "txUnit",
```

Comment:
I placed them only for easier debugging / testing. We already have constants for these but they are UnitType which is uint8 so we don't have their full name as string, only the constant name. Do you think it would make sense to extract all these names to constants ?

Reply:
do we need to use the identifiers in other places? if we don't and it's only for debugging then no need to extract.

Reply:
no, we use them just for better log messages